auth/middleware.go

```diff
@@ -51,33 +52,55 @@ func sessionCallbacks(request *http.Request, writer http.ResponseWriter) *http.R
 }
 
 func reqisterUser(request *http.Request, writer http.ResponseWriter, db *gorm.DB) *http.Request {
```

Comment:
Hey, the code looks fine, could you add some tests in auth/middleware_test.go?

Reply:
I added some HTTP status-code returns and a few tests.